src/ast/passes/semantic_analyser.cpp

```diff
@@ -3324,5 +3355,18 @@ Pass CreateSemanticPass()
   return Pass("Semantic", fn);
 };
 
+void SemanticAnalyser::dereference_if_needed(Expression *&expr)
+{
+  const auto &type = expr->type;
+  if (type.IsRefTy()) {
+    expr = new Unop(Operator::MUL, expr, expr->loc);
```

Comment:
If we convert the wrapped reference here into a pointer, then I think that will simplify later passes by not having to care about reference types any more. I'm hoping we can get away with zero changes to codegen and field_analyser. With luck, it might help with the crash I found too.

Here's an example to explain what I mean a bit better.
Original AST:
```
. :: [reference (struct Foo)]
  builtin: args :: [struct uprobeFunction1_args]
  foo
```

Current transformation:
```
dereference :: [struct Foo]
  . :: [reference (struct Foo)]
    builtin: args :: [struct uprobeFunction1_args]
    foo
```

AST after converting reference into pointer:
```
dereference :: [struct Foo]
  . :: [pointer (struct Foo)]
    builtin: args :: [struct uprobeFunction1_args]
    foo
```

Reply:
Brilliant!